src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpEnvironmentProxy.cs

```diff
@@ -168,18 +178,73 @@ private static Uri GetUriFromString(string value)
             {
                 return null;
             }
+            if (value.StartsWith("http://", StringComparison.OrdinalIgnoreCase))
+            {
+                value = value.Substring(7);
```

Comment:
nit: similar to the other nit about allocations, likely can be avoided here too. I will stop pointing those.